### PR TITLE
Fix customer archive bug

### PR DESCRIPTION
**Pull request by FourSigma**
Story details: https://app.shortcut.com/replicated/story/88051/replicated-cli-not-archiving-customers-by-name

This PR fixes a bug in `customer archive` command.  The bug does not accept customer name for the `archive` command.  This fix  allows the command to accept either the customer name or id as a parameter.

* Archive by customer name works: *
```
╭─siva@Sivas-Air ~/repos/replicatedhq/replicated/cli ‹siva/sc-88051/replicated-cli-not-archiving-customers-by●› 
╰─$ ./replicated customer ls      
                     ID                                             NAME             CHANNELS    EXPIRES    TYPE
2V2ayAUFgU58qnGNBQqEUBpMZr9    MyBigCustomer     Beta       Never      prod

╭─siva@Sivas-Air ~/repos/replicatedhq/replicated/cli ‹siva/sc-88051/replicated-cli-not-archiving-customers-by●› 
╰─$ ./replicated customer archive --customer="MyBigCustomer"    
                                                                                       
╭─siva@Sivas-Air ~/repos/replicatedhq/replicated/cli ‹siva/sc-88051/replicated-cli-not-archiving-customers-by●› 
╰─$ ./replicated customer ls                                
ID    NAME    CHANNELS    EXPIRES    TYPE
```

* Archive by customer id works * (no regression)
```
╭─siva@Sivas-Air ~/repos/replicatedhq/replicated/cli ‹siva/sc-88051/replicated-cli-not-archiving-customers-by●› 
╰─$ ./replicated customer ls                                
ID                                                                     NAME                   CHANNELS    EXPIRES                          TYPE
2V2c8OQUzs6jXaDb6mvU9n8vQ3u    MySecondBigCustomer     Beta       2023-09-20 21:57:08 +0000 UTC    prod
╭─siva@Sivas-Air ~/repos/replicatedhq/replicated/cli ‹siva/sc-88051/replicated-cli-not-archiving-customers-by●› 
╰─$ ./replicated customer archive --customer="2V2c8OQUzs6jXaDb6mvU9n8vQ3u"
  ╭─siva@Sivas-Air ~/repos/replicatedhq/replicated/cli ‹siva/sc-88051/replicated-cli-not-archiving-customers-by●› 
╰─$ ./replicated customer ls                                              
ID    NAME    CHANNELS    EXPIRES    TYPE
```

### CODE REVIEW

File: cli/cmd/runner.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ type runnerArgs struct {
 
 	customerLsIncludeTest bool
 
+	archiveCustomerNameOrId           string
```
Comment (bbawn) [inline cli/cmd/runner.go:80]: Nit, for consistency with existing `nounVerbObject`names:
```suggestion
	customerArchiveNameOrId           string
```

**Review by bbawn — CHANGES_REQUESTED** [review]
Looks good, one naming nit.

Also, one more nit: we conventionally put a link to the ShortCut story permalink in the first line of the PR. Reverse link to PR is then automatically added to shortcut.

File: cli/cmd/runner.go (hunk 1 above)
Comment (FourSigma) [reply cli/cmd/runner.go]: Done!

**Comment by FourSigma** [comment]
> Looks good, one naming nit.
> 
> Also, one more nit: we conventionally put a link to the ShortCut story permalink in the first line of the PR. Reverse link to PR is then automatically added to shortcut.

Done

**Review by bbawn — APPROVED** [review]

**Pull request merged**